### PR TITLE
Replace `org.kohsuke:pom` with `io.jenkins.tools:central-repository-pom`

### PR DESCRIPTION
**Pull request by NotMyFault**
Closes https://github.com/jenkinsci/archetypes/pull/646

Thanks for approving my OSSRH issue!

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -61,18 +70,6 @@
...
-                    <artifactId>maven-javadoc-plugin</artifactId>
-                    <version>3.5.0</version>
-                </plugin>
-                <plugin>
-                    <groupId>org.sonatype.plugins</groupId>
-                    <artifactId>nexus-staging-maven-plugin</artifactId>
-                    <configuration>
-                        <autoReleaseAfterClose>true</autoReleaseAfterClose>
-                    </configuration>
-                </plugin>
```
Comment (NotMyFault) [inline pom.xml:75]: That's part of the updated parent pom

Comment (jglick) [reply pom.xml:75]: Could probably also upstream https://github.com/jenkinsci/archetypes/blob/00a88f9ecec7cca427f3500e3c1055586d7bc894/pom.xml#L83 & https://github.com/jenkinsci/archetypes/blob/00a88f9ecec7cca427f3500e3c1055586d7bc894/pom.xml#L88-L92 (do not recall original reasons for these overrides) and https://github.com/jenkinsci/archetypes/blob/00a88f9ecec7cca427f3500e3c1055586d7bc894/pom.xml#L67

**Review by jglick — APPROVED** [review]

**Comment by jglick** [comment]
@NotMyFault I presume that by now you have permission to merge to this repo yourself, and cut releases?

File: pom.xml (hunk 1 above)
Comment (NotMyFault) [reply pom.xml:75]: Good idea, I've moved them in 1.1

**Pull request merged**